pallets/author-inherent/src/lib.rs

```diff
@@ -222,3 +223,112 @@ impl<T: Config> ProvideInherent for Module<T> {
 		Ok(())
 	}
 }
+
+#[cfg(test)]
+mod tests {
+	use super::*;
+
+	use frame_support::{
+		assert_err, assert_ok, impl_outer_event, impl_outer_origin, parameter_types,
+		traits::{OnFinalize, OnInitialize},
+	};
+	use sp_core::H256;
+	use sp_io::TestExternalities;
+	use sp_runtime::{
+		testing::Header,
+		traits::{BlakeTwo256, IdentityLookup},
+	};
+
+	pub fn new_test_ext() -> TestExternalities {
+		let t = frame_system::GenesisConfig::default()
+			.build_storage::<Test>()
+			.unwrap();
+		TestExternalities::new(t)
+	}
+
+	impl_outer_origin! {
+		pub enum Origin for Test where system = frame_system {}
+	}
+
+	mod author_inherent {
+		pub use super::super::*;
+	}
+
+	impl_outer_event! {
+		pub enum MetaEvent for Test {
+			frame_system<T>,
+			author_inherent<T>,
+		}
+	}
+
+	impl<T> EventHandler<T> for () {
+		fn note_author(_author: T) {}
+	}
+
+	#[derive(Clone, Eq, PartialEq)]
+	pub struct Test;
+	parameter_types! {
+		pub const BlockHashCount: u64 = 250;
+	}
+	impl System for Test {
+		type BaseCallFilter = ();
+		type BlockWeights = ();
+		type BlockLength = ();
+		type DbWeight = ();
+		type Origin = Origin;
+		type Index = u64;
+		type BlockNumber = u64;
+		type Call = ();
+		type Hash = H256;
+		type Hashing = BlakeTwo256;
+		type AccountId = u64;
+		type Lookup = IdentityLookup<Self::AccountId>;
+		type Header = Header;
+		type Event = MetaEvent;
+		type BlockHashCount = BlockHashCount;
+		type Version = ();
+		type PalletInfo = ();
+		type AccountData = ();
+		type OnNewAccount = ();
+		type OnKilledAccount = ();
+		type SystemWeightInfo = ();
+		type SS58Prefix = ();
+	}
+	impl Config for Test {
+		type Event = MetaEvent;
+		type EventHandler = ();
+		type CanAuthor = ();
+	}
+	type AuthorInherent = Module<Test>;
+	type Sys = frame_system::Module<Test>;
+
+	pub fn roll_to(n: u64) {
+		while Sys::block_number() < n {
+			Sys::on_finalize(Sys::block_number());
+			Sys::set_block_number(Sys::block_number() + 1);
+			Sys::on_initialize(Sys::block_number());
+			AuthorInherent::on_initialize(Sys::block_number());
+		}
+	}
+
+	#[test]
+	fn set_author_works() {
+		new_test_ext().execute_with(|| {
+			assert_ok!(AuthorInherent::set_author(Origin::none(), 1));
+			roll_to(1);
+			assert_ok!(AuthorInherent::set_author(Origin::none(), 1));
+			roll_to(2);
+		});
+	}
+
+	#[test]
+	fn double_author_fails() {
+		new_test_ext().execute_with(|| {
+			assert_ok!(AuthorInherent::set_author(Origin::none(), 1));
+			assert_err!(
```

Comment:
```suggestion
			assert_noop!(
```

https://crates.parity.io/frame_support/macro.assert_noop.html

> Evaluate an expression, assert it returns an expected Err value and that runtime storage has not been mutated (i.e. expression is a no-operation).